### ISSUE TITLE
Flat tree analysis

### ISSUE THREAD

**Issue by sidnarayanan** [sidnarayanan/PandaCore#8]
Should gradually move all flat-tree manipulation to `rootpy`/`root_numpy`-based selection

- Create a base `Selector` class that acts as a wrapper around `root_numpy.root2array` or `root_numpy.tree2array`. Can be used for flat-tree skimming, plotting (as in #7), etc

- Check which C++ utilities can be moved to python using the above model, e.g. `Normalizer`, `BranchAdder`, etc. Perhaps the C++ methods should be maintained (slower, but less memory-intensive) in case they are needed.

- Replace C++ `TMVA` tools with common interface to ML libraries (`TMVA`, `scikit-learn` to start, eventually `tensorflow`)

**Comment by sidnarayanan** [comment]
The first point is implemented in #13 , albeit as a set of functions that manipulate structured arrays. Still need to consider if a class solution is more useful. For example, could deal with field title mapping in a more transparent way without modifying the underlying structured array.

**Comment by sidnarayanan** [comment]
In regards to point two: I think there is not a good way to add a branch to an existing tree in `rootpy`. Two options are:
- Add a friend `TTree`. But how do I save to the file that the first tree should friend the new tree without needing to rewrite the first tree? This might also break other functionality
- Read the entire source tree into memory, compute the new branch, and write a new tree. This cannot be more efficient than the existing implementation, in which only relevant branches are read.

I think for now, we cannot replace the existing C++ implementations without sacrificing speed. Point three may make me rethink this though, as we'd need a way of informing the C++ object of a python method to classify new inputs.
